### PR TITLE
Add rubygem-bcrypt package

### PR DESCRIPTION
**Pull request by lzap**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly

WIP I will blindly try to abuse our CI if I am lucky.

---

### CODE REVIEW

File: packages/foreman/rubygem-bcrypt/rubygem-bcrypt.spec [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+%{gem_libdir}
+%exclude %{gem_cache}
+%{gem_spec}
+
+%files doc
+%doc %{gem_docdir}
+%exclude %{gem_instdir}/.rspec
+%doc %{gem_instdir}/CHANGELOG
+%{gem_instdir}/Gemfile
+%{gem_instdir}/Gemfile.lock
```
Comment (ekohl) [inline packages/foreman/rubygem-bcrypt/rubygem-bcrypt.spec]: Should this be excluded?

File: packages/foreman/rubygem-bcrypt/rubygem-bcrypt.spec [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+rm -rf %{buildroot}%{gem_instdir}/ext/
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/ext
+%{gem_extdir_mri}
+%exclude %{gem_instdir}/.gitignore
+%exclude %{gem_instdir}/.travis.yml
+%license %{gem_instdir}/COPYING
+%{gem_instdir}/appveyor.yml
```
Comment (ekohl) [inline packages/foreman/rubygem-bcrypt/rubygem-bcrypt.spec]: Can be excluded

File: packages/foreman/rubygem-bcrypt/rubygem-bcrypt.spec [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+mkdir -p %{buildroot}%{gem_extdir_mri}
+cp -a .%{gem_extdir_mri}/{gem.build_complete,*.so} %{buildroot}%{gem_extdir_mri}/
+
+# Prevent dangling symlink in -debuginfo (rhbz#878863).
+rm -rf %{buildroot}%{gem_instdir}/ext/
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/ext
```
Comment (ekohl) [inline packages/foreman/rubygem-bcrypt/rubygem-bcrypt.spec]: Since this is removed with `rm -rf` in `%install` I think this is causing the build failure.

**Comment by lzap** [comment]
Green, it works:

```
[root@sat65 ~]# scl enable tfm -- irb
irb(main):001:0> require "bcrypt"
=> true
irb(main):002:0> BCrypt::Engine.generate_salt(5)
=> "$2a$05$Sp7LB6ziNqdtYxfKSP8uTu"
irb(main):003:0> BCrypt::Engine.hash_secret("test", BCrypt::Engine.generate_salt(5))
=> "$2a$05$fHlLLkj5sz1TtBBECNCHbOcnZVpG.xFVWQyVhiaTuckhTOrlIZFWm"
```

**Pull request merged**